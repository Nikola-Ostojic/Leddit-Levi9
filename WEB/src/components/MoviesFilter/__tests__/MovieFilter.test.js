import React from "react";
import { shallow } from "enzyme";
import { Button, TextField } from "@material-ui/core";
import Select from "@material-ui/core/Select";

import MoviesFilter from "../MoviesFilter";

describe("MoviesFilter component tests", () => {
  const mockedProps = {
    totalPages: 3,
    movieName: "Test",
    setMovieName: jest.fn(),
    currentPage: 2,
    setCurrentPage: jest.fn(),
    itemsPerPage: 10,
    setItemsPerPage: jest.fn()
  };
  it("Should match snapshot", () => {
    const wrapper = shallow(<MoviesFilter {...mockedProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("Should trigger set movie name on text field on change event", () => {
    const wrapper = shallow(<MoviesFilter {...mockedProps} />);
    const eventMock = {
      target: {
        value: mockedProps.movieName
      }
    };
    wrapper.find(TextField).simulate("change", eventMock);

    expect(mockedProps.setMovieName).toHaveBeenCalledWith(
      mockedProps.movieName
    );
  });

  it("Should decrement currentPage on first button click", () => {
    const wrapper = shallow(<MoviesFilter {...mockedProps} />);

    wrapper.find(Button).at(0).simulate("click");

    expect(mockedProps.setCurrentPage).toHaveBeenCalledWith(
      mockedProps.currentPage - 1
    );
  });

  it("Should increment currentPage on second button click", () => {
    const wrapper = shallow(<MoviesFilter {...mockedProps} />);

    wrapper.find(Button).at(1).simulate("click");

    expect(mockedProps.setCurrentPage).toHaveBeenCalledWith(
      mockedProps.currentPage + 1
    );
  });

  it("Should call setItemsPerPage on dropdown change", () => {
    const wrapper = shallow(<MoviesFilter {...mockedProps} />);
    const eventMock = {
      target: {
        value: 20
      }
    };
    wrapper.find(Select).simulate("change", eventMock);

    expect(mockedProps.setItemsPerPage).toHaveBeenCalledWith(
      eventMock.target.value
    );
  });
});
