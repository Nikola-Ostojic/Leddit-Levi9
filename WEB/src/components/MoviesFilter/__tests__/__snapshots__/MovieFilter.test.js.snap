// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesFilter component tests Should match snapshot 1`] = `
<div
  className="filterContainer"
>
  <WithStyles(ForwardRef(TextField))
    fullWidth={true}
    onChange={[Function]}
    placeholder="Search"
    value="Test"
  />
  <div
    className="pagingContainer"
  >
    <div>
      <WithStyles(ForwardRef(InputLabel))>
        Page
      </WithStyles(ForwardRef(InputLabel))>
      <div>
        <WithStyles(ForwardRef(Button))
          disabled={false}
          onClick={[Function]}
        >
          <ArrowBackIcon />
        </WithStyles(ForwardRef(Button))>
        2
         / 
        3
        <WithStyles(ForwardRef(Button))
          disabled={false}
          onClick={[Function]}
        >
          <ArrowForwardIcon />
        </WithStyles(ForwardRef(Button))>
      </div>
    </div>
    <div>
      <WithStyles(ForwardRef(FormControl))
        className="formControl"
      >
        <WithStyles(ForwardRef(InputLabel))>
          Page size
        </WithStyles(ForwardRef(InputLabel))>
        <WithStyles(ForwardRef(Select))
          onChange={[Function]}
          value={10}
        >
          <WithStyles(ForwardRef(MenuItem))
            value={1}
          >
            1
          </WithStyles(ForwardRef(MenuItem))>
          <WithStyles(ForwardRef(MenuItem))
            value={5}
          >
            5
          </WithStyles(ForwardRef(MenuItem))>
          <WithStyles(ForwardRef(MenuItem))
            value={10}
          >
            10
          </WithStyles(ForwardRef(MenuItem))>
          <WithStyles(ForwardRef(MenuItem))
            value={20}
          >
            20
          </WithStyles(ForwardRef(MenuItem))>
          <WithStyles(ForwardRef(MenuItem))
            value={30}
          >
            30
          </WithStyles(ForwardRef(MenuItem))>
        </WithStyles(ForwardRef(Select))>
      </WithStyles(ForwardRef(FormControl))>
    </div>
  </div>
</div>
`;
